rust-runtime/aws-smithy-observability/src/meter.rs

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Metrics are used to gain insight into the operational performance and health of a system in
+//! real time.
+
+use crate::attributes::{Attributes, Context};
+use crate::error::ObservabilityError;
+
+/// Provides named instances of [Meter].
+pub trait MeterProvider {
```

Comment:
Traits in heavily used runtime components typically have super traits like `Send + Sync + Debug` (ex. [HttpClient](https://github.com/smithy-lang/smithy-rs/blob/8cf9ebddd8cf6793e80a52e9b7c883cc3c1e89d6/rust-runtime/aws-smithy-runtime-api/src/client/http.rs#L135)) to make consuming types be sent & shared among threads. Curious to know about new traits introduced in this PR in that regard 🤔

Reply:
Good call, initially I was going to hold off on adding these and add them as-needed just to minimize what the user might need to implement. But looking at your example and others it seems we tend to be fairly proactive with those supertraits so I added them in https://github.com/smithy-lang/smithy-rs/pull/3912/commits/fe50a9dd499faa70ec4be9ad3bcc43bd13785b4c